platforms/jvm/ear/src/main/java/org/gradle/plugins/ear/descriptor/internal/DefaultDeploymentDescriptor.java

```diff
@@ -451,4 +404,18 @@ private Object nodeNameFor(String name) {
     XmlTransformer getTransformer() {
         return transformer;
     }
+
+    public DefaultDeploymentDescriptor copyFrom(DeploymentDescriptor descriptor) {
+        this.fileName = descriptor.getFileName();
+        getVersion().set(descriptor.getVersion());
+        getApplicationName().set(descriptor.getApplicationName());
+        getInitializeInOrder().set(descriptor.getInitializeInOrder());
+        getDescription().set(descriptor.getDescription().getOrNull());
+        getDisplayName().set(descriptor.getDisplayName().getOrNull());
+        getLibraryDirectory().set(descriptor.getLibraryDirectory());
+        getModules().set(descriptor.getModules());
+        getSecurityRoles().set(descriptor.getSecurityRoles());
+        getModuleTypeMappings().set(descriptor.getModuleTypeMappings());
+        return this;
+    }
```

Comment:
:x: This looks like part of an anti-pattern, and would be good if we could delete.

Reply:
I was able to avoid that with this change: [44e4b5a](https://github.com/gradle/gradle/pull/30496/commits/44e4b5aa3a2fc59c2bcd4b28541cc100a30336cf)

But I think we should merge that directly to master to have smaller changes on EAP branch.